cmd/pinniped/cmd/kubeconfig_test.go

```diff
@@ -2850,7 +2850,7 @@ func TestGetKubeconfig(t *testing.T) {
 			})
 			issuerEndpointPtr = &issuerEndpoint
 
-			testLog := testlogger.New(t)
+			testLog := testlogger.NewLegacy(t) //nolint: staticcheck  // old test with lots of log statements
```

Comment:
You mentioned on a call that the changes to the logs are pretty small and inconsequential, so it makes more sense to just continue using the old format rather than changing hundreds of test assertions. How will we know if in the future there is a change that's more consequential that we should pick up?

Reply:
> You mentioned on a call that the changes to the logs are pretty small and inconsequential

Specifically, some white space changes in the logs and the ordering of keys is slightly different now.  More than enough to break all of our test assertions in a way that is non-trivial to fix.

> so it makes more sense to just continue using the old format rather than changing hundreds of test assertions.

Only for old, preexisting tests.

> How will we know if in the future there is a change that's more consequential that we should pick up?

Two things:

1. I marked `testlogger.NewLegacy` as deprecated so our linter will get upset if it is used.  This is to discourage its use in new tests (they should use `testlogger.New` instead).  This will give us coverage for both the old and new format.
2. None of this actually matters for the real logs our code emits because those logs use the `klog` format, which is distinctly different from the `stdr` log format.  I do not know why we chose this format for our tests, but it is what we have today (and it does assert that certain logs happened at certain times).  As part of the audit work, I would like to migrate this stuff in a way that it is consistent with the real logs being emitted by our code (which should be some kind of JSON format at that point).